ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/policy-maps-qos.j2

```diff
@@ -18,5 +18,24 @@ policy-map type quality-of-service {{ policy_map.name }}
       set {{ cli_set }} {{ class.set[set] }}
 {%             endif %}
 {%         endfor %}
+{%         if class.police.cir is arista.avd.defined %}
+{%             if class.police.cir.cir_rate is arista.avd.defined and class.police.cir.cir_unit is arista.avd.defined and class.police.cir.bc is arista.avd.defined and class.police.cir.bc_unit is arista.avd.defined %}
+      police cir {{ class.police.cir.cir_rate }} {{ class.police.cir.cir_unit }} bc {{ class.police.cir.bc }} {{ class.police.cir.bc_unit }}
```

Comment:
your schema shows default values for units, so you should not check for defined. Instead use arista.avd.default filter on the values when inserting in the string.

Reply:
Sure, thanks